### ISSUE TITLE
formatter: percentage keyframe block inside a declaration-shaped rule fails to parse after #26194

### ISSUE THREAD

**Issue by miloag** [oxc-project/oxc#26291]
### What happens

#26194 made a declaration followed by a block parse and format. It works when the nested blocks are `from { }` / `to { }`, but a percentage keyframe selector inside the same shape still fails.

```css
@animate {
  blink: blink 1100ms linear infinite {
    50% {
      opacity: 0;
    }
  }
}
```

```
  x Syntax error: a top-level `{}` block is disallowed in a declaration value
   ,-[src/index.css:2:39]
 1 | @animate {
 2 |   blink: blink 1100ms linear infinite {
   :                                       ^
 3 |     50% {
   `----
```

The same file with `to { opacity: 0; }` in place of `50% { opacity: 0; }` formats fine in 0.66.0.

Prettier 3.9.6 (`--parser css`) formats the percentage version unchanged.

### Related

Follow-up to #26158 / #26194. The shape is [Rainbow Index](https://github.com/rainbowindex/rainbowindex)'s `@animate` directive, whose body is a `@keyframes` body.

### Versions

- oxfmt 0.66.0
- prettier 3.9.6 for the comparison
- macOS, Node 24
